### PR TITLE
Create Tabs stories

### LINKED ISSUE #334: Add Tabs component to Storybook
Like in the title, also make sure that the visuals match what's in Figma: https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3198%3A6107

### PR DESCRIPTION
**Pull request by drillprop**
Fix #334

### CODE REVIEW

File: src/shared/components/Tabs/Tabs.styles.tsx [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+  width: 100%;
+`
+
+export const TabsGroup = styled.div`
+  display: flex;
+  position: relative;
+  scroll-behavior: smooth;
+  overflow: auto;
+  ::-webkit-scrollbar {
+    display: none;
```
Comment (mikkio-j) [inline src/shared/components/Tabs/Tabs.styles.tsx:25]: scrollbar in Firefox is still visible, so maybe you could add something like `scrollbar-width: none;` there

File: src/shared/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1,29 +1,64 @@
...
+              setSelected(idx)
+            }}
+            key={`${tab}-${idx}`}
+            selected={selected === idx}
+          >
+            <span>{tab}</span>
+          </Tab>
+        ))}
+      </TabsGroup>
+      {isRightGradientVisible && <BackgroundGradient direction="next" />}
```
Comment (kdembler) [inline src/shared/components/Tabs/Tabs.tsx]: I see we only display a right shadow which isn't ideal. If we use a shadow as an indication of "some more content is there" then we will need it on both sides once you scroll. It also should disappear when you scroll all the way to the right

File: src/shared/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -1,29 +1,112 @@
...
+      if (!tabsRef.current || !width) {
+        return
+      }
+      const isScrollBarVisible = tabsRef.current.scrollWidth > width
+      if (isScrollBarVisible) {
+        setCenter(width / 3)
+      } else {
+        setCenter(0)
+      }
+    },
```
Comment (kdembler) [inline src/shared/components/Tabs/Tabs.tsx]: Could you explain a bit what's happening here? I don't understand

Comment (kdembler) [reply src/shared/components/Tabs/Tabs.tsx]: I guess I don't get what `center` is supposed to represent

File: src/shared/components/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -1,29 +1,112 @@
...
+  })
+
+  useEffect(() => {
+    const tabsGroup = tabsRef.current
+    if (!tabsGroup) {
+      return
+    }
+    const currentItemOffsetleft = selected > 0 && TAB_WIDTH * selected
```
Comment (kdembler) [inline src/shared/components/Tabs/Tabs.tsx]: I think we could simplify the flow here if we move the `selected > 0` condition to the `if` above and return if it's `< 1`

File: src/shared/components/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -1,29 +1,112 @@
...
+        timeout={100}
+        classNames={transitions.names.fade}
+        unmountOnExit
+      >
+        <BackgroundGradient direction="prev" />
+      </CSSTransition>
+      <TabsGroup ref={tabsRef}>
+        {tabs.map((tab, idx) => (
+          <Tab
+            onClick={() => {
```
Comment (kdembler) [inline src/shared/components/Tabs/Tabs.tsx]: I know this is carried from the old code, but maybe we could elevate that function to a named handler to make it more readable? We can use the `createClickHandler` pattern

**Review by kdembler — COMMENTED** [review]
Works great. Left some small comments and asked for clarification

File: src/shared/components/Tabs/Tabs.tsx (hunk 3 above)
Comment (drillprop) [reply src/shared/components/Tabs/Tabs.tsx]: This variable here basically means "the desired position of the visible middle Tab" 😅 . I used that value to move scroll a bit if the user clicks on a certain tab - `scrollLeft` will change. 
Maybe I'll rename that to just `middleTabPosition`

Comment (kdembler) [reply src/shared/components/Tabs/Tabs.tsx]: `middleTabPosition` certainly sounds better, we should try to make variable names as descriptive as possible. Now that I know what that is supposed to represent, why do we do `setCenter(width / 3)`?

File: src/shared/components/Tabs/Tabs.tsx [hunk 6]
```diff
@@ -1,29 +1,112 @@
...
+    },
+  })
+
+  useEffect(() => {
+    const tabsGroup = tabsRef.current
+    if (!tabsGroup || selected < 1) {
+      return
+    }
+    const currentItemOffsetleft = TAB_WIDTH * selected
+    if (currentItemOffsetleft) {
```
Comment (kdembler) [inline src/shared/components/Tabs/Tabs.tsx]: I guess we don't need this condition anymore

File: src/shared/components/Tabs/Tabs.tsx [hunk 7]
```diff
@@ -1,29 +1,95 @@
...
+    const { scrollLeft, clientWidth, scrollWidth } = tabsGroup
+
+    const middleTabPosition = clientWidth / 2 - TAB_WIDTH / 2
+    const currentItemOffsetleft = TAB_WIDTH * selected
+
+    tabsGroup.scrollLeft = currentItemOffsetleft - middleTabPosition
+
+    const touchHandler = throttle(() => {
+      setShadowsVisible({
+        left: scrollLeft > SCROLL_SHADOW_OFFSET,
```
Comment (kdembler) [inline src/shared/components/Tabs/Tabs.tsx]: `scrollLeft` here will always be 0 because that's what you create handler with

**Review by kdembler — CHANGES_REQUESTED** [review]
Shadows not working currently

**Review by kdembler — APPROVED** [review]

**Pull request merged**